### DISCUSSION TITLE
OTP Works on Simulator but not on real device(IOS)

### DISCUSSION THREAD

**Discussion by sikloDev** [invertase/react-native-firebase#5969] [General]
"@react-native-firebase/app": "^12.8.0",
    "@react-native-firebase/auth": "^12.8.0",
    "@react-native-firebase/firestore": "^12.8.0",
    "react": "17.0.1",
    "react-native": "0.64.2",

Platform
XCODE 12 -> IOS 14

my code is this

```
const signInWithPhoneNumber = async(phoneNumber) => {
      const confirmation = await auth().signInWithPhoneNumber(phoneNumber);
      setConfirm(confirmation);
    }
```

This perfectly works on an android simulator or real device and also this also works on ios simulator BUT NOT ON THE REAL DEVICE

I have already uploaded the development and production apn on firebase everything is set up. but when i build for a test flight the OTP does not work. I can't receive any SMS with the code in it , on ios simulator it works . It sends SMS on my device . Please do tell me what could be the problem ?

**Comment by mikehardy** [comment]
My first take is "never troubleshoot on old versions", otherwise, I'm not sure what could be wrong, sorry. The fact it works in some builds but not others means the module is working as a given. There is something project-specific wrong with production set up

**Comment by umakantv** [comment]
@mikehardy I am using later versions, but still facing this issue.

```
"@react-native-firebase/app": "^20.1.0",
"@react-native-firebase/auth": "^20.5.0",
"@react-native-firebase/messaging": "^20.1.0",
"react": "18.2.0",
"react-native": "0.74.1",
```

**Reply by mikehardy** [reply]
@umakantv not "later", *latest*. Please test *current* versions - 20.1.0 is old, and 21.7.2 is current. In addition:

```
"@react-native-firebase/app": "^20.1.0",
"@react-native-firebase/auth": "^20.5.0",
"@react-native-firebase/messaging": "^20.1.0",
```

Do not mix and match versions - that is not valid in a react-native-firebase context they need to always match --> https://invertase.io/blog/react-native-firebase-versioning

**Reply by umakantv** [reply]
I updated the version and also updated the `Google-Services-Info.plist` and that fixed the issue.
Thanks
